RepoDb.Core/RepoDb/Extensions/DbConnectionExtension.cs

```diff
@@ -198,7 +198,7 @@ public static async Task<IDbConnection> EnsureOpenAsync(this IDbConnection conne
             {
                 using (var reader = command.ExecuteReader())
                 {
-                    var result = (IEnumerable<dynamic>)DataReader.ToEnumerable(reader, dbFields, connection.GetDbSetting()).AsList();
+                    var result = DataReader.ToEnumerable(reader, dbFields, connection.GetDbSetting());
```

Comment:
I think we still need the `AsList()` on the sync method as the `ToEnumerable()` is a yielding process.